# Incremental Capstone — Session 3

## Data Analysis with Pandas

### Objective

The goal of Session 3 is to analyze the prepared NSMES1988 healthcare dataset using Pandas.

In this session, we will:

- Import the updated dataset created during Session 2
- Identify and report the different data types
- Identify categorical variables
- Perform detailed pivot table analysis using health and region
- Analyze healthcare visits and demographic and economic factors
- Create age and gender distribution tables
- Analyze health status by gender
- Analyze income distribution by gender
- Analyze income distribution across regions
- Analyze the relationship between age and income
- Report the findings and observations from the analysis

In [1]:
# Import Pandas and use the standard alias "pd".
# Pandas will be used to load, organize, group, pivot,
# and analyze the healthcare dataset.
import pandas as pd

# Import NumPy and use the standard alias "np".
# NumPy provides numerical operations that may be useful
# during the analysis.
import numpy as np

## 2. Import the Updated Dataset

The `NSMES1988updated.csv` file created during Session 2 will be imported into a new Pandas DataFrame.

This file contains the prepared dataset, including age represented in actual years and income represented in U.S. dollars.

In [2]:
# Read the updated CSV file created during Session 2.
# The relative path "../data/" moves from the notebooks folder
# to the data folder where the CSV file is stored.
df = pd.read_csv("../data/NSMES1988updated.csv")

# Display the first five rows to verify that
# the dataset loaded successfully.
df.head()

,visits,nvisits,ovisits,novisits,emergency,hospital,health,chronic,adl,region,age,gender,married,school,income,employed,insurance,medicaid
0,5,0,0,0,0,1,average,2,normal,other,69.0,male,yes,6,28810.0,yes,yes,no
1,1,0,2,0,2,0,average,2,normal,other,74.0,female,yes,10,27478.0,no,yes,no
2,13,0,0,0,3,3,poor,4,limited,other,66.0,female,no,10,6532.0,no,no,yes
3,16,0,5,0,1,1,poor,2,limited,other,76.0,male,yes,3,6588.0,no,yes,no
4,3,0,0,0,0,0,average,2,limited,other,79.0,female,yes,6,6588.0,no,yes,no


## 3. Inspect Dataset Structure and Data Types

The structure of the updated dataset will be examined to identify the number of observations, number of variables, non-null values, and data types assigned to each variable.

This inspection will help distinguish numerical data from categorical data before performing the Pandas analysis.

In [3]:
# Display the number of rows and columns in the DataFrame.
# shape returns the result as (rows, columns).
df.shape

(4406, 18)

In [4]:
# Display detailed structural information about the DataFrame.
# info() shows the number of rows, column names, non-null counts,
# data types, and memory usage.
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 4406 entries, 0 to 4405
Data columns (total 18 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   visits     4406 non-null   int64  
 1   nvisits    4406 non-null   int64  
 2   ovisits    4406 non-null   int64  
 3   novisits   4406 non-null   int64  
 4   emergency  4406 non-null   int64  
 5   hospital   4406 non-null   int64  
 6   health     4406 non-null   str    
 7   chronic    4406 non-null   int64  
 8   adl        4406 non-null   str    
 9   region     4406 non-null   str    
 10  age        4406 non-null   float64
 11  gender     4406 non-null   str    
 12  married    4406 non-null   str    
 13  school     4406 non-null   int64  
 14  income     4406 non-null   float64
 15  employed   4406 non-null   str    
 16  insurance  4406 non-null   str    
 17  medicaid   4406 non-null   str    
dtypes: float64(2), int64(8), str(8)
memory usage: 619.7 KB


### Data Type Report

The `NSMES1988updated.csv` dataset contains 4,406 observations and 18 variables. All 18 variables contain 4,406 non-null values, indicating that no missing values are present.

Pandas identified three data types within the dataset:

- Eight variables are stored as `int64`: `visits`, `nvisits`, `ovisits`, `novisits`, `emergency`, `hospital`, `chronic`, and `school`. These variables represent whole-number counts or measurements.
- Two variables are stored as `float64`: `age` and `income`. These variables represent numerical measurements.
- Eight variables are stored as strings: `health`, `adl`, `region`, `gender`, `married`, `employed`, `insurance`, and `medicaid`. These variables contain categorical information represented by text labels.

The dataset therefore contains both numerical and categorical data, which will be analyzed using appropriate Pandas grouping and pivoting techniques.

## 4. Identify Categorical Variables

The dataset contains several variables that describe groups or classifications rather than numerical measurements.

These categorical variables will be examined to identify their unique values before they are used in grouping and pivot table analysis.

In [5]:
# Create a list containing the categorical columns
# identified from the dataset structure.
categorical_columns = [
    "health",
    "adl",
    "region",
    "gender",
    "married",
    "employed",
    "insurance",
    "medicaid"
]

# Display the number of unique values in each
# categorical variable.
df[categorical_columns].nunique()

health       3
adl          2
region       4
gender       2
married      2
employed     2
insurance    2
medicaid     2
dtype: int64

### Categorical Variable Values

The unique values within each categorical variable will be examined to understand the classifications represented in the dataset.

This is particularly important for the `health` and `region` variables because they will be used in the required pivot table analysis.

In [6]:
# Loop through each categorical column.
# For each column, display the column name followed
# by all unique category values found in that variable.
for column in categorical_columns:
    print(f"{column}: {df[column].unique()}")

health: <StringArray>
['average', 'poor', 'excellent']
Length: 3, dtype: str
adl: <StringArray>
['normal', 'limited']
Length: 2, dtype: str
region: <StringArray>
['other', 'midwest', 'northeast', 'west']
Length: 4, dtype: str
gender: <StringArray>
['male', 'female']
Length: 2, dtype: str
married: <StringArray>
['yes', 'no']
Length: 2, dtype: str
employed: <StringArray>
['yes', 'no']
Length: 2, dtype: str
insurance: <StringArray>
['yes', 'no']
Length: 2, dtype: str
medicaid: <StringArray>
['no', 'yes']
Length: 2, dtype: str


### Categorical Data Findings

Eight categorical variables were identified in the dataset: `health`, `adl`, `region`, `gender`, `married`, `employed`, `insurance`, and `medicaid`.

The `health` variable contains three categories: average, poor, and excellent. The `adl` variable contains two categories: normal and limited. The `region` variable contains four categories: other, midwest, northeast, and west. The `gender` variable contains male and female categories.

The `married`, `employed`, `insurance`, and `medicaid` variables each contain two categories represented by yes and no values.

These variables represent classifications rather than numerical measurements and are therefore appropriate for categorical analysis, grouping, and pivot tables.

## 5. Health and Region Pivot Analysis

A pivot table will be created to examine the distribution of individuals across health status and geographic region.

The table will count the number of observations for each combination of `health` and `region`. This provides a clear view of how the health categories are distributed across the four regional categories.

In [9]:
# Create a pivot table showing the number of individuals
# for each combination of health status and region.
health_region_pivot = pd.pivot_table(
    df,
    index="health",
    columns="region",
    values="age",
    aggfunc="count",
    margins=True,
    margins_name="Total"
)

# Improve the labels so the table is easier to read.
health_region_pivot.index.name = "Health Status"
health_region_pivot.columns.name = "Region"

# Display the completed pivot table.
health_region_pivot

Region,midwest,northeast,other,west,Total
Health Status,,,,,
average,957,694,1237,621,3509
excellent,90,57,105,91,343
poor,110,86,272,86,554
Total,1157,837,1614,798,4406


### Health and Region Pivot Findings

The pivot table shows that average health is the most frequently reported health status in every region. The number of individuals reporting average health is 957 in the Midwest, 694 in the Northeast, 1,237 in the Other region, and 621 in the West.

Excellent health is the least frequently reported category in the Midwest, Northeast, and Other regions, with 90, 57, and 105 individuals respectively. In the West, 91 individuals report excellent health compared with 86 reporting poor health.

The Other region contains the largest raw number of observations across all three health categories. However, these values represent counts rather than percentages. Therefore, the larger counts may reflect differences in the total number of individuals represented by each region and should not by themselves be interpreted as evidence that one region has better or worse health.

### Average Visits by Health Status and Region

The next pivot table examines the average number of healthcare visits for individuals within each health status and region combination.

Unlike the previous pivot table, which counted the number of individuals, this analysis calculates the mean number of visits. This allows healthcare utilization to be compared across health status and geographic region.

In [10]:
# Create a pivot table showing the average number of visits
# for each combination of health status and region.
health_region_visits_pivot = pd.pivot_table(
    df,
    index="health",
    columns="region",
    values="visits",
    aggfunc="mean"
)

# Rename the row and column labels to make
# the output easier for a reader to understand.
health_region_visits_pivot.index.name = "Health Status"
health_region_visits_pivot.columns.name = "Region"

# Display the averages rounded to two decimal places.
health_region_visits_pivot.style.format("{:,.2f}")

Region,midwest,northeast,other,west
Health Status,,,,
average,5.28,5.69,5.09,6.50
excellent,3.44,4.02,3.14,3.37
poor,8.12,10.67,8.75,8.59


### Average Visits by Health Status and Region Findings

The pivot table shows a consistent relationship between self-reported health status and average healthcare visits across all four regions.

Individuals reporting poor health have the highest average number of visits in every region. Average visits among individuals reporting poor health range from 8.12 visits in the Midwest to 10.67 visits in the Northeast. The Northeast has the highest average utilization among individuals reporting poor health.

Individuals reporting excellent health have the lowest average number of visits in every region. Average visits for this group range from 3.14 visits in the Other region to 4.02 visits in the Northeast.

Individuals reporting average health fall between the excellent and poor health groups in every region, with average visits ranging from 5.09 in the Other region to 6.50 in the West.

Overall, the results show that poorer self-reported health is associated with greater healthcare utilization in this dataset. Regional differences are also present, although health status appears to show a consistent relationship with the average number of visits across all regions.

## 6. Analysis of Healthcare Visits and Demographic Factors

The dataset contains several variables representing different forms of healthcare utilization. These variables will first be summarized to understand their distributions before examining their relationships with demographic and economic factors such as gender, marital status, education, income, employment, insurance, and Medicaid.

In [11]:
# Create a list containing the healthcare utilization variables.
visit_columns = [
    "visits",
    "nvisits",
    "ovisits",
    "novisits",
    "emergency",
    "hospital"
]

# Calculate descriptive statistics for the visit variables.
visit_summary = df[visit_columns].describe().T

# Keep the statistics that are most useful for this analysis.
visit_summary = visit_summary[
    ["count", "mean", "50%", "std", "min", "max"]
]

# Rename the columns to make the table easier to read.
visit_summary.columns = [
    "Count",
    "Mean",
    "Median",
    "Std. Dev.",
    "Minimum",
    "Maximum"
]

# Give the row labels a descriptive name.
visit_summary.index.name = "Visit Type"

# Display the results with two decimal places.
visit_summary.style.format("{:,.2f}")

,Count,Mean,Median,Std. Dev.,Minimum,Maximum
Visit Type,,,,,,
visits,"4,406.00",5.77,4.00,6.76,0.00,89.00
nvisits,"4,406.00",1.62,0.00,5.32,0.00,104.00
ovisits,"4,406.00",0.75,0.00,3.65,0.00,141.00
novisits,"4,406.00",0.54,0.00,3.88,0.00,155.00
emergency,"4,406.00",0.26,0.00,0.70,0.00,12.00
hospital,"4,406.00",0.30,0.00,0.75,0.00,8.00


### Healthcare Visit Summary Findings

All six healthcare utilization variables contain 4,406 observations, confirming that no missing values are present in these measures.

The `visits` variable has the highest average utilization at 5.77 visits, with a median of 4 visits. The maximum value of 89 visits indicates that some individuals have substantially greater healthcare utilization than the typical individual.

The remaining visit variables have medians of zero. `nvisits` has a mean of 1.62, while `ovisits` and `novisits` have means of 0.75 and 0.54 respectively. Emergency visits and hospital stays have the lowest averages at 0.26 and 0.30.

The zero medians indicate that at least half of the individuals recorded no utilization for `nvisits`, `ovisits`, `novisits`, emergency visits, and hospital stays. At the same time, maximum values such as 104 `nvisits`, 141 `ovisits`, and 155 `novisits` show that a smaller number of individuals have very high healthcare utilization.

Overall, the differences between the means, medians, and maximum values suggest that several healthcare utilization variables have right-skewed distributions, where many individuals have low or zero utilization and a smaller group accounts for much higher utilization.

### Healthcare Utilization by Gender

Healthcare utilization will be grouped by gender to compare the average number of visits across the six healthcare utilization measures.

This analysis helps determine whether the observed average utilization differs between male and female individuals in the dataset.

In [12]:
# Group the dataset by gender and calculate the mean
# for each healthcare utilization variable.
visits_by_gender = df.groupby("gender")[visit_columns].mean()

# Improve the table labels for presentation.
visits_by_gender.index.name = "Gender"

# Rename the visit columns to make them easier to read.
visits_by_gender.columns = [
    "Visits",
    "Nvisits",
    "Ovisits",
    "Novisits",
    "Emergency",
    "Hospital"
]

# Display the average values rounded to two decimal places.
visits_by_gender.style.format("{:,.2f}")

,Visits,Nvisits,Ovisits,Novisits,Emergency,Hospital
Gender,,,,,,
female,6.01,1.73,0.68,0.51,0.27,0.29
male,5.42,1.45,0.85,0.57,0.25,0.31


### Healthcare Utilization by Gender Findings

The analysis shows some differences in average healthcare utilization between female and male individuals.

Females have a higher average number of overall visits, with 6.01 visits compared with 5.42 visits for males. Females also have a higher average number of `nvisits`, at 1.73 compared with 1.45 for males.

Males have higher average `ovisits` and `novisits`, with averages of 0.85 and 0.57 compared with 0.68 and 0.51 for females. Average hospital utilization is also slightly higher among males at 0.31 compared with 0.29 for females.

Emergency utilization is very similar between the two groups, averaging 0.27 for females and 0.25 for males.

Overall, the results indicate modest differences in healthcare utilization by gender. These descriptive results show associations within the dataset but do not establish that gender causes the observed differences.

### Healthcare Utilization by Marital Status

Healthcare utilization will be grouped by marital status to compare the average number of visits between married and unmarried individuals.

This analysis helps identify whether healthcare utilization patterns differ according to marital status within the dataset.

In [13]:
# Group the dataset by marital status and calculate
# the mean for each healthcare utilization variable.
visits_by_marital = df.groupby("married")[visit_columns].mean()

# Improve the table labels for presentation.
visits_by_marital.index.name = "Marital Status"

# Rename the columns to make the visit types easier to read.
visits_by_marital.columns = [
    "Visits",
    "Nvisits",
    "Ovisits",
    "Novisits",
    "Emergency",
    "Hospital"
]

# Display the averages rounded to two decimal places.
visits_by_marital.style.format("{:,.2f}")

,Visits,Nvisits,Ovisits,Novisits,Emergency,Hospital
Marital Status,,,,,,
no,5.93,1.55,0.65,0.51,0.30,0.31
yes,5.64,1.68,0.83,0.56,0.23,0.28


### Healthcare Utilization by Marital Status Findings

Healthcare utilization is relatively similar between married and unmarried individuals, although several differences are present.

Unmarried individuals have a slightly higher average number of overall visits at 5.93 compared with 5.64 for married individuals. They also have higher average emergency utilization at 0.30 compared with 0.23 and slightly higher hospital utilization at 0.31 compared with 0.28.

Married individuals have higher average `nvisits`, `ovisits`, and `novisits`. Married individuals average 1.68 `nvisits`, 0.83 `ovisits`, and 0.56 `novisits`, compared with 1.55, 0.65, and 0.51 respectively among unmarried individuals.

Overall, the differences in average healthcare utilization by marital status are relatively modest. The results suggest some variation in the types of healthcare services used by married and unmarried individuals, but the descriptive analysis alone does not establish a causal relationship.

### School Distribution

The `school` variable represents years of schooling and ranges from 0 to 18 years. Unlike binary categorical variables such as gender or marital status, school is a numerical variable with multiple possible values.

The distribution of school years will first be examined to determine how individuals are distributed across the different education levels before comparing education with healthcare utilization.

In [14]:
# Count the number of individuals at each reported
# number of years of schooling.
school_distribution = df["school"].value_counts().sort_index()

# Convert the Series into a DataFrame so the output
# can be given clearer, presentation-ready labels.
school_distribution = school_distribution.to_frame(
    name="Number of Individuals"
)

# Give the row index a descriptive label.
school_distribution.index.name = "Years of School"

# Display counts using comma separators and no decimal places.
school_distribution.style.format(
    {"Number of Individuals": "{:,.0f}"}
)

,Number of Individuals
Years of School,
0,103
1,13
2,38
3,71
4,100
5,103
6,173
7,221
8,684


### School Distribution Findings

The number of years of schooling ranges from 0 to 18 years, with substantial differences in the number of individuals represented at each education level.

Twelve years of schooling is the most common value, representing 1,252 individuals. Eight years of schooling is the second most common value, with 684 individuals. Several other education levels contain considerably fewer observations.

The concentration at 12 years may reflect individuals completing approximately a high-school level of education, although the dataset only identifies the variable as years of schooling and does not directly provide educational credentials.

Because the number of individuals differs substantially across schooling levels, comparisons should consider these differences in group size when interpreting healthcare utilization.

### Average Healthcare Visits by Years of Schooling

The relationship between education and healthcare utilization will be examined by calculating the average number of healthcare visits for each reported number of years of schooling.

The number of individuals in each education group will also be included so that differences in group size can be considered when interpreting the average number of visits.

In [15]:
# Group the dataset by years of schooling.
# Calculate both the number of individuals and
# the average number of healthcare visits in each group.
school_visits = df.groupby("school").agg(
    Individuals=("visits", "count"),
    Average_Visits=("visits", "mean")
)

# Give the row index a descriptive label.
school_visits.index.name = "Years of School"

# Rename the columns for a cleaner presentation.
school_visits.columns = [
    "Number of Individuals",
    "Average Visits"
]

# Display counts as whole numbers and averages
# rounded to two decimal places.
school_visits.style.format({
    "Number of Individuals": "{:,.0f}",
    "Average Visits": "{:,.2f}"
})

,Number of Individuals,Average Visits
Years of School,,
0,103,4.21
1,13,5.54
2,38,5.08
3,71,6.24
4,100,4.75
5,103,5.85
6,173,5.36
7,221,5.45
8,684,5.58


### School and Healthcare Utilization Findings

Average healthcare visits vary across years of schooling, and the relationship is not perfectly linear.

Individuals with 18 years of schooling have the highest average number of visits at 7.89, followed by individuals with 15 years of schooling at 7.08. Several of the higher schooling levels have average visit counts above the overall dataset average of approximately 5.77 visits.

However, the average number of visits does not consistently increase with every additional year of schooling. For example, individuals with 16 years of schooling average 6.09 visits, which is lower than the averages for 15, 17, and 18 years.

Group sizes also differ substantially. Twelve years of schooling contains the largest group with 1,252 individuals, while some education levels contain relatively few observations. For example, only 13 individuals report one year of schooling.

Overall, the descriptive results suggest some variation in healthcare utilization across education levels, with several higher schooling levels showing greater average utilization. However, the pattern is not strictly linear, and differences in group size should be considered when interpreting the results.

### Healthcare Utilization by Employment Status

Healthcare utilization will be grouped by employment status to compare average healthcare use between employed and unemployed individuals.

This analysis will help identify whether healthcare utilization patterns differ according to employment status within the dataset.

In [16]:
# Group the dataset by employment status and calculate
# the mean for each healthcare utilization variable.
visits_by_employment = df.groupby("employed")[visit_columns].mean()

# Give the row index a descriptive label.
visits_by_employment.index.name = "Employment Status"

# Rename the columns to make the output easier to read.
visits_by_employment.columns = [
    "Visits",
    "Nvisits",
    "Ovisits",
    "Novisits",
    "Emergency",
    "Hospital"
]

# Display the averages rounded to two decimal places.
visits_by_employment.style.format("{:,.2f}")

,Visits,Nvisits,Ovisits,Novisits,Emergency,Hospital
Employment Status,,,,,,
no,5.81,1.64,0.79,0.55,0.27,0.30
yes,5.49,1.43,0.44,0.42,0.22,0.23


### Healthcare Utilization by Employment Status Findings

The analysis shows that individuals who are not employed have higher average healthcare utilization across all six healthcare measures compared with employed individuals.

Individuals who are not employed average 5.81 overall visits compared with 5.49 visits among employed individuals. They also have higher average `nvisits` at 1.64 compared with 1.43 and higher `ovisits` at 0.79 compared with 0.44.

The same pattern is present for `novisits`, emergency visits, and hospital utilization. Individuals who are not employed average 0.55 `novisits`, 0.27 emergency visits, and 0.30 hospital visits, compared with 0.42, 0.22, and 0.23 respectively among employed individuals.

Overall, individuals who are not employed show greater average healthcare utilization across every visit category examined. However, these descriptive results show an association between employment status and healthcare utilization and do not establish that employment status causes the differences.

### Healthcare Utilization by Insurance Status

Healthcare utilization will be grouped by insurance status to compare the average use of healthcare services between insured and uninsured individuals.

This analysis will help identify whether healthcare utilization patterns differ according to insurance coverage within the dataset.

In [17]:
# Group the dataset by insurance status and calculate
# the mean for each healthcare utilization variable.
visits_by_insurance = df.groupby("insurance")[visit_columns].mean()

# Give the row index a descriptive label.
visits_by_insurance.index.name = "Insurance Status"

# Rename the columns to make the output easier to read.
visits_by_insurance.columns = [
    "Visits",
    "Nvisits",
    "Ovisits",
    "Novisits",
    "Emergency",
    "Hospital"
]

# Display the averages rounded to two decimal places.
visits_by_insurance.style.format("{:,.2f}")

,Visits,Nvisits,Ovisits,Novisits,Emergency,Hospital
Insurance Status,,,,,,
no,4.91,0.93,1.06,0.44,0.33,0.31
yes,6.02,1.81,0.66,0.57,0.24,0.29


### Healthcare Utilization by Insurance Status Findings

Healthcare utilization differs between insured and uninsured individuals, although the direction of the difference varies by type of healthcare service.

Insured individuals have a higher average number of overall visits at 6.02 compared with 4.91 among uninsured individuals. Insured individuals also have higher average `nvisits`, at 1.81 compared with 0.93, and higher average `novisits`, at 0.57 compared with 0.44.

In contrast, uninsured individuals have higher average `ovisits`, at 1.06 compared with 0.66 among insured individuals. Uninsured individuals also have higher average emergency utilization at 0.33 compared with 0.24 and slightly higher average hospital utilization at 0.31 compared with 0.29.

Overall, insurance status is associated with differences in healthcare utilization patterns. Insured individuals show greater utilization in some categories, while uninsured individuals show greater utilization in others. These descriptive results identify associations within the dataset but do not establish that insurance coverage causes the observed differences.

### Healthcare Utilization by Medicaid Status

The dataset's `medicaid` variable will be used to examine the Medical Aid factor identified in the project requirements.

Healthcare utilization will be grouped by Medicaid status to compare the average use of healthcare services between individuals with and without Medicaid coverage.

In [18]:
# Group the dataset by Medicaid status and calculate
# the mean for each healthcare utilization variable.
visits_by_medicaid = df.groupby("medicaid")[visit_columns].mean()

# Give the row index a descriptive label.
visits_by_medicaid.index.name = "Medicaid Status"

# Rename the columns to make the output easier to read.
visits_by_medicaid.columns = [
    "Visits",
    "Nvisits",
    "Ovisits",
    "Novisits",
    "Emergency",
    "Hospital"
]

# Display the averages rounded to two decimal places.
visits_by_medicaid.style.format("{:,.2f}")

,Visits,Nvisits,Ovisits,Novisits,Emergency,Hospital
Medicaid Status,,,,,,
no,5.68,1.65,0.73,0.53,0.25,0.28
yes,6.71,1.26,1.01,0.55,0.44,0.42


### Healthcare Utilization by Medicaid Status Findings

Healthcare utilization differs between individuals with and without Medicaid coverage.

Individuals with Medicaid have a higher average number of overall visits at 6.71 compared with 5.68 among individuals without Medicaid. Medicaid recipients also have higher average `ovisits` at 1.01 compared with 0.73, higher emergency utilization at 0.44 compared with 0.25, and higher hospital utilization at 0.42 compared with 0.28.

Average `novisits` are similar between the two groups, at 0.55 for individuals with Medicaid and 0.53 for individuals without Medicaid.

The primary exception to the overall pattern is `nvisits`. Individuals without Medicaid average 1.65 `nvisits`, compared with 1.26 among individuals with Medicaid.

Overall, individuals with Medicaid show greater average utilization across most of the healthcare measures examined. However, these results describe an association within the dataset and do not establish that Medicaid coverage causes greater healthcare utilization.

## 7. Age and Gender Distribution

The distribution of individuals by age and gender will be examined by counting the number of male and female individuals represented at each age.

This analysis provides a demographic view of the dataset and allows differences in age distribution between genders to be observed.

In [26]:
# Create the age and gender distribution table.
age_gender_distribution = pd.crosstab(
    df["age"],
    df["gender"],
    margins=True,
    margins_name="Total"
)

# Improve table labels.
age_gender_distribution.index.name = "Age"
age_gender_distribution.columns.name = "Gender"

# Format counts with commas and display ages as whole numbers.
age_gender_distribution.style.format(
    "{:,.0f}"
).format_index(
    lambda value: (
        "Total" if value == "Total"
        else str(int(float(value)))
    ),
    axis=0
)

Gender,female,male,Total
Age,,,
66,190,160,350
67,198,140,338
68,173,123,296
69,162,131,293
70,174,117,291
71,154,115,269
72,158,123,281
73,156,97,253
74,141,108,249


## 8. Health Status by Gender

A distribution table will be created to examine self-reported health status separately for female and male individuals.

This analysis will count the number of individuals within each health status and gender combination and include totals for easier comparison.

In [27]:
# Create a cross-tabulation showing the number of individuals
# in each health status separated by gender.
health_gender_distribution = pd.crosstab(
    df["health"],
    df["gender"],
    margins=True,
    margins_name="Total"
)

# Give the rows and columns descriptive labels.
health_gender_distribution.index.name = "Health Status"
health_gender_distribution.columns.name = "Gender"

# Display the counts using comma separators.
health_gender_distribution.style.format("{:,.0f}")

Gender,female,male,Total
Health Status,,,
average,"2,093","1,416","3,509"
excellent,193,150,343
poor,342,212,554
Total,"2,628","1,778","4,406"


### Health Status by Gender Findings

Average health is the most commonly reported health status for both females and males. Of the 4,406 individuals in the dataset, 3,509 report average health, including 2,093 females and 1,416 males.

Poor health is reported by 554 individuals, including 342 females and 212 males. Excellent health is the least common category, with 343 individuals, including 193 females and 150 males.

Females have higher raw counts than males in all three health categories. However, the dataset contains more females overall, with 2,628 females compared with 1,778 males. Therefore, the higher female counts should not by themselves be interpreted as evidence that females are more likely to report a particular health status.

Overall, average health is the dominant self-reported health category for both genders, while excellent health is the least frequently reported category.

## 9. Income Distribution by Gender

Income will be analyzed separately for female and male individuals using the number of observations, mean income, median income, minimum income, and maximum income.

Both mean and median income are included because the earlier descriptive analysis showed that income is right-skewed. The median can therefore provide useful context when unusually high income values influence the mean.

In [28]:
# Group the dataset by gender and calculate
# descriptive statistics for income.
income_by_gender = df.groupby("gender")["income"].agg(
    Count="count",
    Mean="mean",
    Median="median",
    Minimum="min",
    Maximum="max"
)

# Give the row index a descriptive label.
income_by_gender.index.name = "Gender"

# Rename the columns for presentation.
income_by_gender.columns = [
    "Number of Individuals",
    "Mean Income",
    "Median Income",
    "Minimum Income",
    "Maximum Income"
]

# Display counts as whole numbers and income as currency.
income_by_gender.style.format({
    "Number of Individuals": "{:,.0f}",
    "Mean Income": "${:,.2f}",
    "Median Income": "${:,.2f}",
    "Minimum Income": "${:,.2f}",
    "Maximum Income": "${:,.2f}"
})

,Number of Individuals,Mean Income,Median Income,Minimum Income,Maximum Income
Gender,,,,,
female,"2,628","$22,493.48","$14,160.00","$-10,125.00","$548,351.00"
male,"1,778","$29,377.16","$20,574.00","$-10,125.00","$548,351.00"
